scheduler/src/cook/mesos/api.clj

```diff
@@ -1071,6 +1146,28 @@
             (render-jobs-for-response conn framework-id retrieve-sandbox-directory-from-agent ctx)))]
     (read-jobs-handler conn is-authorized-fn {:handle-ok handle-ok})))
 
+(defn read-instances-handler
+  [conn is-authorized-fn resource-attrs]
+  (base-cook-handler
+    (merge {:allowed-methods [:get]
+            :malformed? instance-request-malformed?
+            :allowed? (partial instance-request-allowed? conn is-authorized-fn)
+            :exists? (partial instances-exist? conn)}
+           resource-attrs)))
+
+(defn read-instances-handler-multiple
+  [conn framework-id is-authorized-fn retrieve-sandbox-directory-from-agent]
+  (let [handle-ok (partial render-instances-for-response conn framework-id retrieve-sandbox-directory-from-agent)]
+    (read-instances-handler conn is-authorized-fn {:handle-ok handle-ok})))
+
+(defn read-instances-handler-single
+  [conn framework-id is-authorized-fn retrieve-sandbox-directory-from-agent]
+  (let [handle-ok
+        (fn handle-ok [ctx]
+          (first
+            (render-instances-for-response conn framework-id retrieve-sandbox-directory-from-agent ctx)))]
```

Comment:
This might be a bit more symmetrical with the `-multiple` version above:
```clojure
(->> (partial render-instances-for-response ...)
     (comp first))
```

Reply:
Done